wp/lib/bap_wp/src/environment.ml

```diff
@@ -78,6 +79,8 @@ and fun_spec = {
   spec : fun_spec_type
 }
 
+and indirect_spec = t -> Constr.t -> Constr.t * t
```

Comment:
I think it might be a good idea to include the `Exp.t` as an argument in case we want to use that information in the spec in the future.